client/my-sites/email/titan-mail-add-mailboxes/index.jsx

```diff
@@ -0,0 +1,305 @@
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import Main from 'calypso/components/main';
+import QueryProductsList from 'calypso/components/data/query-products-list';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import { recordTracksEvent as recordTracksEventAction } from 'calypso/state/analytics/actions';
+import {
+	TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+	TITAN_MAIL_MONTHLY_SLUG,
+} from 'calypso/lib/titan/constants';
+import TitanExistingForwardsNotice from 'calypso/my-sites/email/titan-mail-add-mailboxes/titan-existing-forwards-notice';
+import TitanMailboxPricingNotice from 'calypso/my-sites/email/titan-mail-add-mailboxes/titan-mailbox-pricing-notice';
+import { titanMailMonthly } from 'calypso/lib/cart-values/cart-items';
+import TitanNewMailboxList from 'calypso/my-sites/email/titan-mail-add-mailboxes/titan-new-mailbox-list';
+import TitanUnusedMailboxesNotice from 'calypso/my-sites/email/titan-mail-add-mailboxes/titan-unused-mailbox-notice';
+
+import { withLocalizedMoment } from 'calypso/components/localized-moment';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class TitanMailAddMailboxes extends React.Component {
+	state = {
+		mailboxes: [ buildNewTitanMailbox( this.props.selectedDomainName, false ) ],
+	};
+
+	isMounted = false;
+
+	componentDidMount() {
+		this.isMounted = true;
+	}
+
+	componentWillUnmount() {
+		this.isMounted = false;
+	}
+
+	recordClickEvent = ( eventName, eventProps ) => {
+		const { recordTracksEvent, selectedDomainName } = this.props;
+		recordTracksEvent( eventName, {
+			...eventProps,
+			domain_name: selectedDomainName,
+		} );
+	};
+
+	goToEmail = () => {
+		const {
+			currentRoute,
+			isSelectedDomainNameValid,
+			selectedDomainName,
+			selectedSite,
+		} = this.props;
+
+		page(
+			emailManagement(
+				selectedSite.slug,
+				isSelectedDomainNameValid ? selectedDomainName : null,
+				currentRoute
+			)
+		);
+	};
+
+	handleCancel = () => {
+		this.recordClickEvent( 'calypso_email_management_titan_add_mailboxes_cancel_button_click' );
+		this.goToEmail();
+	};
+
+	getCartItem = () => {
+		const { maxTitanMailboxCount, selectedDomainName } = this.props;
+		const mailboxes = this.state.mailboxes;
+		const quantity = mailboxes.length + maxTitanMailboxCount;
+		const new_quantity = mailboxes.length;
+		const email_users = mailboxes.map( transformMailboxForCart );
+
+		return titanMailMonthly( {
+			domain: selectedDomainName,
+			quantity,
+			extra: {
+				email_users,
+				new_quantity,
+			},
+		} );
+	};
+
+	handleContinue = () => {
+		const { selectedSite } = this.props;
+		const { mailboxes } = this.state;
+		const canContinue = areAllMailboxesValid( mailboxes, [ 'alternativeEmail' ] );
+
+		this.recordClickEvent( 'calypso_email_management_titan_add_mailboxes_continue_button_click', {
+			can_continue: canContinue,
+			mailbox_count: mailboxes.length,
+		} );
+
+		if ( canContinue ) {
+			this.props.shoppingCartManager
```

Comment:
Can we add a busy state to the appropriate button when we're waiting for the shopping cart to be updated? I've seen this happen in other places after switching to the shopping cart manager and it leads to a very bad UX where it seems nothing is happening once the user clicks the button.

Reply:
Done in [this commit](https://github.com/Automattic/wp-calypso/pull/51217/commits/786072241c80faef8d69561d9edfd85da16d434c).